packages/flutter/lib/src/widgets/dual_transition_builder.dart

```diff
@@ -0,0 +1,200 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+/// Builder callback used by [DualTransitionBuilder].
+///
+/// The builder is expected to return a transition powered by the provided
+/// `animation` and wrapping the provided `child`.
+///
+/// The `animation` provided to the builder always runs forward from 0.0 to 1.0.
+typedef AnimatedTransitionBuilder = Widget Function(
+  BuildContext context,
+  Animation<double> animation,
+  Widget child,
+);
+
+/// A transition builder that animates its [child] based on the
```

Comment:
We should consider what this means for the copy of `DualTransitionBuilder` in the animations package once this makes it into stable. Would we be able to safely remove `DualTransitionBuilder` from the animations package and use this version instead? 

Also, I don't think this'll be a problem, but I'll ask just in case since I'm not sure what will actually happen -- Will there be any conflicts/problems with having both the copy here and the copy in the animations package exist at the same time?

cc/ @goderbauer

Reply:
We should delete it from the animations package and use the one from the framework here. Deleting itself should be fine since the builder was never part of the public interface of the animations package. However while the DualTransitionBuilder exists in two places, I suspect there may be a conflict within the animations package as there are now two competing DualTransitionBuilder definitions. If that is the case, we should hide the framework implementation in the package until this change has made it to stable and then delete the implementation in the package (while increasing the minimal flutter version). 

Can you please check whether the example app in the animation package builds correctly with this change applied to your Flutter version? If it doesn't we will have to hide this implementation in the package before this change can land.

Reply:
> Can you please check whether the example app in the animation package builds correctly with this change applied to your Flutter version? If it doesn't we will have to hide this implementation in the package before this change can land.

Got it, scheduled for tomorrow since it's already 2.A.M.

Reply:
@goderbauer Confirmed, there're conflicts in animations package if we export it.

![image](https://user-images.githubusercontent.com/15884415/83935217-2305f400-a7ea-11ea-9a5f-350bfdc55bca.png)